app/main/posts/common/post-actions.directive.js

```diff
@@ -51,9 +51,12 @@ function PostActionsDirective(
                     $location.path().indexOf('collections') === -1 &&
                     $location.path().indexOf('savedsearches') === -1) {
                     // Redirect to list
-                    $location.path('/views/data');
+                    $state.go('posts.data');
+                } else if ($state.current.name === 'posts.data' || $state.current.name === 'posts.data.detail') {
+                    // we only want to do this in the data detail view because it's where you can really update a list
+                    $rootScope.$broadcast('event:edit:post:status:data:mode:saveSuccess', {post: $scope.post, deleted: true});
                 } else {
-                    $state.go('posts.data', null, { reload: true }); // in favor of $route.reload();
+                    $state.reload();
```

Comment:
changed to $state.reload to emulate https://github.com/ushahidi/platform-client/blob/225ecb43baef0011cb995b53b26e5dc12670c210/app/main/posts/common/post-actions.directive.js#L58

+ kept the broadcast because that's what was being used for archive to work in the old data view/etc but wasn't there for delete .

`Works in all views. `